qcommon/src/org/qcmg/common/util/FileUtils.java

```diff
@@ -61,8 +62,11 @@ public static String getFileCheckSum( final String file ) {
 	        MessageDigest md = MessageDigest.getInstance("MD5");
 	        FileChannel channel = inputStream.getChannel();
 	        ByteBuffer buff = ByteBuffer.allocate(2048);
+	
 	        while(channel.read(buff) != -1){
-	            buff.flip();
+	        	//jdk8 don't have java.nio.ByteBuffer.flip()
+	        	//we have to cast it to Buffer, so it can run JDK8 and above
+	            ((Buffer)buff).flip();
```

Comment:
you don't need to cast here - `ByteBuffer` is a subclass of `Buffer` and so can see its public and protected methods.
(just like your call to `clear()`)

Reply:
Just made our tool more flexible, it can run on both JDK8 and above

Reply:
Leaving the cast out should work fine in java8

Reply:
To cast  ByteBuffer to Buffer before calling flip(), Then users can compile the project with JDK9+ and have it running on JDK8.